pkg/commands/status_discovery.go

```diff
@@ -53,19 +61,51 @@ func (s *CloudDiscoveryStatusStore) GetApiIngressStatus(cluster *kops.Cluster) (
 
 	if awsCloud, ok := cloud.(awsup.AWSCloud); ok {
 		name := "api." + cluster.Name
-		lb, err := awstasks.FindLoadBalancerByNameTag(awsCloud, name)
-		if lb == nil {
-			return nil, nil
+
+		ELB, NLB := false, false
+
+		lbSpec := cluster.Spec.API.LoadBalancer
+		switch lbSpec.Class {
+		case kops.LoadBalancerClassClassic, "":
+			ELB = true
+		case kops.LoadBalancerClassNetwork:
+			NLB = true
+		default:
+			return nil, fmt.Errorf("Unknown Cluster.Spec.API.LoadBalancer.Class : %v", lbSpec.Class)
 		}
-		if err != nil {
-			return nil, fmt.Errorf("error looking for AWS ELB: %v", err)
+
+		var lb interface{}
+
+		if ELB {
```

Comment:
I suggest simplifying, not having separate `ELB` and `NLB` booleans. Just check `if cluster.Spec.API.LoadBalancer == kops.LoadBalancerClassNetwork { ... } else { ... }`

Or `switch cluster.Spec.API.LoadBalancer.Class`

Reply:
I think the right strategy is to search both ELB and NLB's